### PR TITLE
[#2604]Dependency management split

### LINKED ISSUE #2606: [Improve][build]  root's pom.xml don't need to manage all dependencies
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In the future, we will support multiple versions of the flink/spark engine, and version management will become an important issue.

seatunnel manages all dependencies in root's pom.xml, while many connectors require different versions of dependencies;
I expect to only manage basic core dependencies such as log4j, commons in root's pom.xml;
The dependencies of connectors and engines are managed by their modules;

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CalvinKirs**
#2604 subtask
Split Spark connector

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -157,17 +156,12 @@
...
         <elasticsearch6.client.version>6.3.1</elasticsearch6.client.version>
         <elasticsearch7.client.version>7.5.1</elasticsearch7.client.version>
         <flink-shaded-hadoop-2.version>2.7.5-7.0</flink-shaded-hadoop-2.version>
         <parquet-avro.version>1.10.0</parquet-avro.version>
-        <elasticsearch-spark.version>6.8.3</elasticsearch-spark.version>
         <clickhouse-jdbc.version>0.2</clickhouse-jdbc.version>
-        <hbase-spark.version>1.0.0</hbase-spark.version>
-        <kudu-spark.version>1.7.0</kudu-spark.version>
-        <mongo-spark.version>2.2.0</mongo-spark.version>
         <spark-redis.version>2.6.0</spark-redis.version>
```
Comment (Hisoka-X) [inline pom.xml]: please move `spark-redis.version` too

File: pom.xml [hunk 2]
```diff
@@ -188,7 +182,6 @@
         <docker.tag>${project.version}</docker.tag>
         <jcommander.version>1.81</jcommander.version>
         <junit.version>5.9.0</junit.version>
-        <tispark.version>2.4.1</tispark.version>
         <druid.version>0.22.1</druid.version>
```
Comment (Hisoka-X) [inline pom.xml:184]: `druid.version` and `sshd.version` also should be moved

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-hudi/pom.xml [hunk 3]
```diff
@@ -28,29 +28,16 @@
...
-            <groupId>org.apache.seatunnel</groupId>
-            <artifactId>seatunnel-api-spark</artifactId>
-            <version>${project.version}</version>
-            <scope>provided</scope>
-        </dependency>
 
         <dependency>
             <groupId>org.apache.hudi</groupId>
             <artifactId>hudi-spark-bundle_${scala.binary.version}</artifactId>
         </dependency>
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-hudi/pom.xml:41]: Add `hudi-spark-bundle.version`

**Comment by CalvinKirs** [comment]
@Hisoka-X Hi, Thanks. PTAL again

**Comment by CalvinKirs** [comment]
image

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**